### PR TITLE
enh(dashboard): enhance arch of dashboard metrics data endpoint

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

this PR intends to enhance the architecture and flow for the Dashboard Metrics Perf Data endpoint

**Fixes** # MON-20804

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Metric/Domain/Model/MetricInformation/RealTimeDataInformation.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+     * Undocumented function.
+     *
+     * @param array<float|null> $values
+     * @param array<array<string>> $labels
+     * @param float|null $minimumValueLimit
+     * @param float|null $maximumValueLimit
+     * @param float|null $minimumValue float handled as string to not lose decimals on *.O value
+     * @param float|null $maximumValue float handled as string to not lose decimals on *.O value
+     * @param privatereadonly?float $lastValue
+     * @param privatereadonly?float $averageValue
```
Comment (jeremyjaouen) [inline centreon/src/Core/Metric/Domain/Model/MetricInformation/RealTimeDataInformation.php:40]:
```suggestion
     * @param float|null $lastValue
     * @param float|null $averageValue
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -17974,4 +17974,10 @@ msgid "Metrics not found"
 msgstr "Métriques non trouvées"
 
 msgid "An error occured while retrieving metrics data"
-msgstr "Une erreur est survenue lors de la récupération des données de métriques"
\ No newline at end of file
+msgstr "Une erreur est survenue lors de la récupération des données de métriques"
+
+msgid "values type in the array are not [%s]"
+msgstr "Le type de valeur dans le tableau ne sont pas [%s]"
```
Comment (jeremyjaouen) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17980]:
```suggestion
```

File: centreon/src/Centreon/Domain/Common/Assertion/AssertionException.php [hunk 3]
```diff
@@ -545,4 +545,9 @@ public static function notJsonEncodable(?string $propertyPath = null): self
             $propertyPath
         );
     }
+
+    public static function invalidTypeInArray(string $type, string $propertyPath): self
+    {
+        return new self(sprintf(_('values type in the array are not [%s]'), $type), self::INVALID_CHOICE, $propertyPath);
```
Comment (jeremyjaouen) [inline centreon/src/Centreon/Domain/Common/Assertion/AssertionException.php:551]:
```suggestion
        return new self(sprintf('values type in the array are not [%s]', $type), self::INVALID_CHOICE, $propertyPath);
```

**Review by smau-centreon — APPROVED** [review]

**Review by dmyios — DISMISSED** [review]

File: centreon/src/Core/Metric/Domain/Model/MetricInformation/DataSource.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Core\Metric\Domain\Model\MetricInformation;
+
+class DataSource
+{
+    public function __construct(
+        private readonly ?int $minimum,
```
Comment (a-launois) [inline centreon/src/Core/Metric/Domain/Model/MetricInformation/DataSource.php:29]: No default values ?

File: centreon/src/Core/Metric/Domain/Model/MetricInformation/GeneralInformation.php [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Core\Metric\Domain\Model\MetricInformation;
+
+class GeneralInformation
+{
+    public function __construct(
+        private readonly int $indexId,
```
Comment (a-launois) [inline centreon/src/Core/Metric/Domain/Model/MetricInformation/GeneralInformation.php:29]: No default values ?

File: centreon/src/Core/Metric/Domain/Model/MetricInformation/MetricInformation.php [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Core\Metric\Domain\Model\MetricInformation;
+
+class MetricInformation
+{
+    public function __construct(
+        private readonly GeneralInformation $generalInformation,
```
Comment (a-launois) [inline centreon/src/Core/Metric/Domain/Model/MetricInformation/MetricInformation.php:29]: No default values ?

File: centreon/src/Core/Metric/Domain/Model/MetricInformation/RealTimeDataInformation.php [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+     * @param array<array<string>> $labels
+     * @param float|null $minimumValueLimit
+     * @param float|null $maximumValueLimit
+     * @param float|null $minimumValue float handled as string to not lose decimals on *.O value
+     * @param float|null $maximumValue float handled as string to not lose decimals on *.O value
+     * @param float|null $lastValue
+     * @param float|null $averageValue
+     */
+    public function __construct(
+        private readonly array $values,
```
Comment (a-launois) [inline centreon/src/Core/Metric/Domain/Model/MetricInformation/RealTimeDataInformation.php:43]: No default values ?

File: centreon/src/Core/Metric/Domain/Model/MetricInformation/RealTimeDataInformation.php [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+namespace Core\Metric\Domain\Model\MetricInformation;
+
+use Centreon\Domain\Common\Assertion\Assertion;
+
+class RealTimeDataInformation
+{
+    /**
+     * Undocumented function.
+     *
```
Comment (a-launois) [inline centreon/src/Core/Metric/Domain/Model/MetricInformation/RealTimeDataInformation.php:32]:
```suggestion
```

File: centreon/src/Core/Metric/Domain/Model/MetricInformation/RealTimeDataInformation.php [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+     * @param array<float|null> $values
+     * @param array<array<string>> $labels
+     * @param float|null $minimumValueLimit
+     * @param float|null $maximumValueLimit
+     * @param float|null $minimumValue float handled as string to not lose decimals on *.O value
+     * @param float|null $maximumValue float handled as string to not lose decimals on *.O value
+     * @param float|null $lastValue
+     * @param float|null $averageValue
+     */
```
Comment (a-launois) [inline centreon/src/Core/Metric/Domain/Model/MetricInformation/RealTimeDataInformation.php:41]: missing `@throws`

File: centreon/src/Core/Metric/Domain/Model/MetricInformation/ThresholdInformation.php [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Core\Metric\Domain\Model\MetricInformation;
+
+class ThresholdInformation
+{
+    public function __construct(
+        private readonly ?float $warningThreshold,
```
Comment (a-launois) [inline centreon/src/Core/Metric/Domain/Model/MetricInformation/ThresholdInformation.php:29]: No default values ?

**Review by a-launois — CHANGES_REQUESTED** [review]

**Review by a-launois — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**